application/apps/rustcore/ts-bindings/spec/session.benchmark.spec.ts

```diff
@@ -0,0 +1,306 @@
+                                            command: `less ${tmpobj2.name}`,
+                                            cwd: process.cwd(),
+                                            envs: process.env as { [key: string]: string },
+                                        })
+                                        .get()
+                                        .sterilized()
+                                );
+                                const shutdownResults = measurement();
+                                const shutdownReport = performanceReport(testName, shutdownResults.ms, test.expectation_ms, `${home_dir}/${test.file}`);
+                                finish(shutdownSession, done, shutdownReport ? undefined : new Error(`${testName} is fail`));
+                                break;
+                            case 'Stream - Open 50 sessions':
+                                const results = [];
+                                const multiSessions = [];
+                                for (let i = 0; i < 50; i++) {
+                                    const file = createSampleFile(
+                                        100,
+                                        logger,
+                                        (j: number) => `file ${i} line data: ${j}\n`
+                                    );
+
+                                    const multiSession = await Session.create();
+                                    multiSessions.push(multiSession);
+
+                                    const multiStream = multiSession.getStream();
+                                    if (multiStream instanceof Error) {
+                                        throw multiStream;
+                                    }
+
+                                    let result = multiStream.observe(
+                                        new Factory.Stream()
+                                            .asText()
+                                            .process({
+                                                command: `less ${file.name}`,
+                                                cwd: process.cwd(),
+                                                envs: process.env as { [key: string]: string },
+                                            })
+                                            .get()
+                                            .sterilized()
+                                    ).catch((err) => `File ${i} failed to open: ${err.message}`);
+                                    results.push(result);
+                                }
+                                const multiResults = measurement();
+                                const multiReport = performanceReport(testName, multiResults.ms, test.expectation_ms, `${home_dir}/${test.file}`);
+                                finish(undefined, done, multiReport ? undefined : new Error(`${testName} is fail`));
+                                break;
```

Comment:
exit from test without closing `session` created on line 42. Also sessions created in this block aren't closed